modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/ClusteredSystemScheduler.java

```diff
@@ -0,0 +1,77 @@
+package com.enonic.xp.impl.scheduler;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.scheduledexecutor.IScheduledExecutorService;
+import com.hazelcast.scheduledexecutor.IScheduledFuture;
+
+import com.enonic.xp.impl.scheduler.distributed.SchedulableTask;
+
+@Component(immediate = true)
+public final class ClusteredSystemScheduler
+    implements SystemScheduler
+{
+    private final IScheduledExecutorService hazelcastExecutor;
+
+    @Activate
+    public ClusteredSystemScheduler( @Reference final HazelcastInstance hazelcastInstance )
+    {
+        hazelcastExecutor = hazelcastInstance.getScheduledExecutorService( "scheduler" );
+    }
+
+    @Override
+    public Set<String> getAllFutures()
+    {
+        return hazelcastExecutor.getAllScheduledFutures().values().
+            stream().
+            flatMap( Collection::stream ).
+            filter( future -> future.getHandler() != null ).
+            map( future -> future.getHandler().getTaskName() ).
+            collect( Collectors.toSet() );
+    }
+
+    @Override
+    public void disposeAllDone()
+    {
+        hazelcastExecutor.getAllScheduledFutures().values().stream().
+            flatMap( Collection::stream ).
+            filter( Future::isDone ).
+            forEach( IScheduledFuture::dispose );
+    }
+
+    @Override
+    public boolean dispose( final String name )
+    {
+        return hazelcastExecutor.getAllScheduledFutures().values().stream().
+            flatMap( Collection::stream ).
+            filter( future -> future.getHandler() != null ).
```

Comment:
I don't think it is ever null.

Reply:
I have thought same before hazelcast gave me proxy object with empty handler.